src/editing_tools/PhotoCanvas.vala

```diff
@@ -29,30 +29,35 @@
 // The PhotoCanvas is an interface object between an EditingTool and its host.  It provides objects
 // and primitives for an EditingTool to obtain information about the image, to draw on the host's
 // canvas, and to be signalled when the canvas and its pixbuf changes (is resized).
-public abstract class EditingTools.PhotoCanvas {
+public abstract class EditingTools.PhotoCanvas : Object {
     public signal void new_surface (Cairo.Context ctx, Dimensions dim);
     public signal void resized_scaled_pixbuf (Dimensions old_dim, Gdk.Pixbuf scaled, Gdk.Rectangle scaled_position);
 
-    public Cairo.Context default_ctx { get; private set; }
-    public Dimensions surface_dim { get; private set; }
-    public Gdk.Pixbuf scaled_pixbuf { get; private set; }
-    public Gdk.Rectangle scaled_position { get; private set; }
-    public Gdk.Window drawing_window { get; private set; }
-    public Gtk.Window container { get; private set; }
-    public Photo photo { get; private set; }
+    public Cairo.Context default_ctx { get; set construct; }
+    public Dimensions surface_dim { get; set construct; }
+    public Gdk.Pixbuf scaled_pixbuf { get; set construct; }
+    public Gdk.Rectangle scaled_position { get; set construct; }
+    public Gdk.Window drawing_window { get; set construct; }
+    public Gtk.Window container { get; set construct; }
+    public Photo photo { get; set construct; }
 
-    private Cairo.Surface scaled;
+    private Cairo.Surface scaled { get; set; }
```

Comment:
```suggestion
    private Cairo.Surface scaled;
```

We don't need to explicit `{ get; set; }` here, this is a private member so it's guaranteed that it can't be accessed out of the class